### DISCUSSION TITLE
Question/Suggestion regarding generation of "alternate version" lists

### DISCUSSION THREAD

**Discussion by HorayNarea** [DandelionSprout/adfilt#400] [Ideas]
To me it looks like the alternate versions of some lists are generated by hand whenever you feel like there where enough changes to the list :D

Am I right with that observation and if yes, would it be possible to have some auto-generation in place so that changes to these lists won't take sometimes multiple days to get into the alternate version?
Currently it's a bit of a hassle when one still has to maintain a custom allowlist with domains from already merged PRs :sweat_smile: 

---

My suggestion would be to create a Github Workflow that auto-runs the generation script whenever a new commit is pushed and host the alternate versions on Github Pages or something

In theory, you could also setup a Workflow that regularly runs PyFunceble on all lists and creates a PR with the changes, kinda like it's already done for the ClearURLs uBo list: https://github.com/DandelionSprout/adfilt/blob/master/.github/workflows/generate_clearurlsubo.yml :thinking:

---

While writing this I thought I could also create a PR that simply implements my suggestions but I could not find a script that generates the alternate versions nor documentation about how you use PyFunceble … I swear I saw at least a script a few months ago but now I can't find it - am I stupid/missremembering or is this your non-shared secret sauce? :laughing:

**Comment by DandelionSprout** [comment]
The lists are predominantly automated by downloading https://raw.githubusercontent.com/DandelionSprout/adfilt/master/NorwegianExperimentalList%20alternate%20versions/XYZPrepareFilters.py and running `Python3.8 XYZPrepareFilters.py` in Cygwin.

However, running the script and uploading the results is currently a manual process.
